services/ui-src/src/components/reports/OverlayModalPage.tsx

```diff
@@ -26,14 +26,20 @@ export const OverlayModalPage = ({
   closeEntityDetailsOverlay,
   route,
 }: Props) => {
-  const { verbiage, modalForm, stepType } = route;
+  const { verbiage, stepType } = route;
   const { report, selectedEntity, setSelectedEntity, editable } = useStore();
   const [selectedStepEntity, setSelectedStepEntity] = useState<
     EntityShape | undefined
   >(undefined);
+  const [selectedEntityIndex, setSelectedEntityIndex] = useState<number>(0);
```

Comment:
All of this code looks pretty good, and runs well too! I think this is the one area I'd like to see an update on. I don't think we need this state object. We have the selectedStepEntity which knows the id thats been selected, and the cards have an objective id that matches from the transformation. We can use that to match the form up.

A function could look something like this: 
```
  const getModalForm = () => {
    if (objectiveCards) {
      for (let card of objectiveCards) {
        if (card.modalForm.objectiveId === selectedStepEntity?.id)
          return card.modalForm;
      }
    } else if (route?.modalForm) {
      return route.modalForm;
    }
    return { id: randomUUID(), formFields: [] };
  };

  let modalForm = getModalForm();
```

That last return is important too - If, for some reason, there isn't an objective card the route won't have a modalForm and thus the page would error out. I don't forsee this happening but it would be good to cover our bases.

Reply:
Could even get rid of that last return since our Typescript is saying a modalForm is guaranteed to exist on the route. So maybe a modalForm should live in the sarJson if nothing but to serve as a empty id and formField to be overwritten.

That would mean the above code looks more like this:
```
  const getModalForm = () => {
    if (objectiveCards) {
      for (let card of objectiveCards) {
        if (card.modalForm.objectiveId === selectedStepEntity?.id)
          return card.modalForm;
      }
    } 
    return route.modalForm;
  };

  let modalForm = getModalForm();
```

Reply:
Great catch, thanks for breaking it down! suggestion has been implemented with a slight modification for selectedStepEntity being undefined at the start,